src/operator/controllers/certmanageradapter/approver.go

```diff
@@ -0,0 +1,125 @@
+package certmanageradapter
+
+import (
+	"context"
+	"fmt"
+	apiutil "github.com/cert-manager/cert-manager/pkg/api/util"
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	utilpki "github.com/cert-manager/cert-manager/pkg/util/pki"
+	"github.com/otterize/credentials-operator/src/controllers/metadata"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// CertificateApprover - This class is inspired by and shares code with https://github.com/cert-manager/csi-driver-spiffe/tree/main/internal/approver/
+type CertificateApprover struct {
+	client    client.Client
+	lister    client.Reader
+	issuerRef cmmeta.ObjectReference
+	registry  *CertManagerWorkloadRegistry
+}
+
+func NewCertificateApprover(issuerRef cmmeta.ObjectReference, client client.Client, lister client.Reader, registry *CertManagerWorkloadRegistry) *CertificateApprover {
+	return &CertificateApprover{
+		issuerRef: issuerRef,
+		client:    client,
+		lister:    lister,
+		registry:  registry,
+	}
+}
+
+func (a *CertificateApprover) Register(ctx context.Context, mgr manager.Manager) error {
+	log := logrus.WithFields(logrus.Fields{"name": "cert-request-aprrover"})
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(new(cmapi.CertificateRequest)).
+		WithEventFilter(predicate.NewPredicateFuncs(func(obj client.Object) bool {
+			var req cmapi.CertificateRequest
+			err := a.lister.Get(ctx, client.ObjectKeyFromObject(obj), &req)
+			if apierrors.IsNotFound(err) {
+				// Ignore CertificateRequests that have been deleted.
+				return false
+			}
+
+			// If an error happens here and we do nothing, we run the risk of not
```

Comment:
To avoid this problem, these conditions should be checked in the Reconciler, which can return errors and retry on failure.